### PR TITLE
Implement automated partition testing

### PR DESCRIPTION
**Pull request by danpaul000**
#### Problem
We currently have no automated way to test cluster stability in a partitioned network, or to detect regressions of the same.

#### Summary of Changes
Implement netem config for forced partitioning/re-integration into the testnet automation framework and run certain testcases as part of the nightly regression suite.

### CODE REVIEW

File: system-test/partition-testcases/gce-5-node-3-partition.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+steps:
+  - command: "system-test/testnet-automation.sh"
+    label: "GCE - CPU Only 5 Node - Partial Loss 3 Partitions"
```
Comment (mvines) [inline system-test/partition-testcases/gce-5-node-3-partition.yml]: Can we run this test on colo too?

Comment (danpaul000) [reply system-test/partition-testcases/gce-5-node-3-partition.yml]: Yes, see the `colo-3-partitions.yml` file.  Each .yml file refers to a single test case configuration, ie, a specific provider, duration, number of nodes, and in this case, what kind of partitions to hit it with.

Comment (mvines) [reply system-test/partition-testcases/gce-5-node-3-partition.yml]: Sorry I guess what I meant was, "let's not run this on GCE and instead use the colo hardware"

Comment (mvines) [reply system-test/partition-testcases/gce-5-node-3-partition.yml]: keep the burn down

**Pull request merged**